Relicensing update post - towards LGPLv2.1 (#325)
MDAnalysis/MDAnalysis.github.io: _posts/2023-09-22-licensing-update.md

Title: An update on relicensing the MDAnalysis library (towards LGPLv2.1+)

> > ⚠️ **Disclaimer**

Reviewer: See https://github.com/orgs/community/discussions/16925

> > ⚠️ **Disclaimer**

Reviewer 2: While this looks very nice on GitHub, it doesn't render so nicely for me on Jekyll -- @RMeli do you know if the final product will have the warning formatting or if it'll look more like the screenshot below?

<img width="718" alt="Screenshot 2023-09-24 at 11 53 52 pm" src="https://github.com/MDAnalysis/MDAnalysis.github.io/assets/31115101/de01e0a9-ff63-4a85-92db-fb51ef57801a">

> > ⚠️ **Disclaimer**

Author: I'm reverting this change - it would be nice to work this out, but we're pretty short on time atm, so let's test this one out next time.

> > ⚠️ **Disclaimer**

Reviewer: Yup, sorry for the brain fart. I just default to thinking in GitHub terms these days.

> represent MDAnalysis' full legal position on software licensing; it
> simply aims to inform MDAnalysis developers and users on certain
> aspects of the MDAnalysis relicensing efforts and how we think it
> may impact them going forward.
>
> Further information on open-source software licensing can be found
> from sources such as the [Open Source Initiative][OSI],
> [tl;drLegal][tldr legal] and the [Software Sustainability Institute][SSI licensing].
>
> Should you have any concerns about licensing, we always strongly
> recommend getting legal advice before making any decisions on how
> licensing changes may affect you.

## Short description

This blog post provides an update on MDAnalysis’ proposed license change. Specifically, we outline our
decision to opt for a [GNU Lesser General Public License v2.1+ (LGPL v2.1+)][LGPL v2.1] instead of
the [previously communicated]() [GNU Lesser General Public License v3+ (LGPL v3+)][LGPL v3].

We also detail some [interim changes](#why-the-interim-gpl-v3-and-lgpl-v3-package-licenses-the-issue-with-apache-20)
to the MDAnalysis [package license](#package-license), alongside some

> [upcoming actions for library contributors](#package-license).

Reviewer 3: I assume this link meant to point somewhere else? #how-does-this-affect-me, maybe?

Author: Switched to #next-steps

## Overview

> In November of 2022 the MDAnalysis team [announced a proposal to change the license of its core library](https://www.mdanalysis.org/2022/11/07/relicensing/#relicensing-mdanalysis)

Reviewer 4: If you can use the Jekyll code that links to posts — something like post_url? That makes the site more robust

from the [GNU General Public License v2+][GPL v2] to the [GNU Lesser General Public License v3+][LGPL v3].
The full rationale for this change can be found [here](https://www.mdanalysis.org/2022/11/07/relicensing/#rationale-for-license-change)
and mainly centered around three main arguments:

1. Allowing downstream codes to freely import or link to the MDAnalysis library with minimal worry about the license they choose to use.
2. Allowing downstream codes to use and subclass any MDAnalysis components under its application programming interface

>    (classes, methods, and data objects), without incurring copyleft and incurring the license choice of the downstream code.

Reviewer 3: ```suggestion
   (classes, methods, and data objects), without incurring copyleft and restricting the license choice of the downstream code.
```

3. Ensuring that codes that either directly copy or extend the MDAnalysis library’s source code would fall under the copyleft

>    license requirements of the MDAnalysis library license. Doing so would ensure that eventual license would more closely match the

Reviewer 3: ```suggestion
   license requirements of the MDAnalysis library license. Doing so would ensure that the eventual license would more closely match the
```

   original intent of the historical [GPL v2+](GPL v2) contributions which were kindly donated to MDAnalysis over the years.

> It is our view that a weak copyleft license, such as the [GNU Lesser General Public License][LGPL v3] would best match these requirements.

Reviewer 4: It was

Author: Isn't still "is"? I.e. weak copyleft is still good, but LGPLv3 is just not the best?

Upon announcing the proposed license change, we engaged in a [consultation period](https://www.mdanalysis.org/2022/11/07/relicensing/#consultation-period-7th-november-until-5th-december-2022)
where we encouraged members of the community to comment on and ask questions about the proposed relicensing efforts.

> Overall, the community response was overwhelmingly positive, and led to some very fruitful discussions. We would like

Author: They are on discord and X + mailing list - I don't want to add them here though.

> Overall, the community response was overwhelmingly positive, and led to some very fruitful discussions. We would like

Reviewer: Ok, no problem.

to thank all members of the community who participated. Some interesting points were brought to our attention, particularly
when it came to the range of licenses compatible with [LGPL v3+][LGPL v3]. This raised a lot of questions about the
suitability of [LGPL v3+][LGPL v3], and the current status of the MDAnalysis [GPL v2+][GPL v2] package license, so we
decided to pause the relicensing process whilst we worked with our legal counsel to find an appropriate path forward.

After careful consideration we have decided to aim for a license change to the [LGPL v2.1+][LGPL v2.1] via interim
[package license](#package-license) changes to [GPL v3+][GPL v3] and [LGPL v3+][LGPL v3].

In this post we will provide our rationale for this change, alongside key information for how we will be
enacting these changes.

**License changes will affect how users and developers will interact with the MDAnalysis code base going

> forward. We urge all members of the community to read this post and [get in touch](#package-license) with

Reviewer 3: ```suggestion
forward. We urge all members of the community to read this post and [get in touch](#getting-in-touch) with
```

the MDAnalysis core developer team should you have any questions or concerns**.

## Rationale for our change in target license

*Table 1: MDAnalysis’ view of the dynamic linking license compatibility matrix. License pairs
detailed with a compatibility of “No” have claimed incompatibilities between their license terms.
License pairs detailed as “Yes ([copyleft](#copyleft])” are compatible but may incur copyleft through the
interacting GNU GPL license. License pairs detailed with the compatibility “Yes” are, to our knowledge,

> fully compatible without incurring any [copyleft](#copyleft) penalty.*

Reviewer 4: I wouldn’t call it “penalty” — just state the fact without adding an opinion — others might not see it as a penalty

|-----------------|---------------------|---------------------|---------------------|---------------------|--------------------|--------------------|--------------------|
| [GPL v2][]      | Yes                 | No \[1\]            | Yes ([copyleft]())  | No \[1\]            | No                 | Yes ([copyleft]()) | Yes ([copyleft]()) |
| [GPL v3][]      | No \[1\]            | Yes                 | No \[2\]            | Yes ([copyleft]())  | Yes ([copyleft]()) | Yes ([copyleft]()) | Yes ([copyleft]()) |
| [LGPL v2.1][]   | Yes ([copyleft]())  | No                  | Yes                 | No \[2\]            | No                 | Yes                | Yes                |
| [LGPL v3][]     | No \[1\]            | Yes ([copyleft]())  | No \[2\]            | Yes                 | Yes                | Yes                | Yes                |
| [Apache 2.0][]  | No                  | Yes ([copyleft]())  | No                  | Yes                 | Yes                | Yes                | Yes                |
| [MIT][]         | Yes ([copyleft]())  | Yes ([copyleft]())  | Yes                 | Yes                 | Yes                | Yes                | Yes                |

> | [BSD 3.0][]     | Yes ([copyleft]())  | Yes ([copyleft]())  | Yes                 | Yes                 | Yes                | Yes                | Yes                |

Reviewer 3: I don't think the `[copyleft]()` in this table will end up linking anywhere - might need to be `[copyleft](#copyleft)`?

*\[1\] Compatible if the [GPL v2][] license is provided under the “or any later version” terms.
\[2\] Compatible if the [LGPL v2.1][] license is provided under the “or any later version” terms.*

### Why LGPL v2.1+ instead of LGPLv3+?

During the community consultation, it was brought to our attention that by choosing an [LGPL v3+][LGPL v3]

> license, would no longer be compatible with older versions of the GNU General Public License, namely

Reviewer 3: ```suggestion
license, it would no longer be compatible with older versions of the GNU General Public License, namely
```

[GPL v2][] and [LGPL v2.1][] (see Table 1). The reasons for this are complex and we will not attempt to
explain them here, we will instead refer the reader to the [FSF GPL license compatibility FAQ entry](https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility)
for further details.

Overall the impact of this incompatibility is likely very low. Few codes are solely licensed under
[GPLv2][] or [LGPLv2.1][], with most under “or any later version” terms, allowing the codes to relicense
under a later version and avoiding this issue entirely. However, this would still require several downstream
packages to update their [package license](#package-license), which is not an insignificant amount of work.
It is our view that it is best for the community that we try to avoid this issue, especially given that the

> MDAnalysis library itself has historically been released under the [GPL v2+][GPL v2].

Reviewer 3: ```suggestion
MDAnalysis library itself has historically been released under [GPL v2+][GPL v2].
```

In this section, we will briefly go over the various steps we propose to take (Table 2) as we go about relicensing the
[MDAnalysis core library][].

### 1. A GPL v2+ package and source code (MDAnalysis v2.5 and lower)

This is the original state of the license up until the [2.5.x releases of MDAnalysis](https://www.mdanalysis.org/blog/page2/#releases-2-5-0-and-2-4-x-of-mdanalysis).
The package and source code were released under a [GPL v2+][GPL v2] license.

### 2. A GPL v3+ package, and a mix of historical GPL v2+ and new LGPL v2.1+ source code (MDAnalysis v2.6 onwards)

As of commit [44733fc][], all new contributions to MDAnalysis have been made under the terms of the
[LGPL v2.1+][LGPL v2.1] and the package is released under [GPL v3+][GPL v3] to better reflect our
understanding of the [Apache 2.0 incompatibility issues](#why-the-interim-gpl-v3-and-lgpl-v3-package-licenses-the-issue-with-apache-20).
Historical source code contributions remain under the terms of [GPL v2+][GPL v2].

This will remain the state of MDAnalysis until we obtain approval from our developers to relicense

> their code to [LGPL v2.1+][LGPL v2.1]. As we obtain contributor agreements, these historical

Reviewer: How is this obtained?

Author: I'll link to the section below

Reviewer: This comment and the above are on the right place. The comments below are shifted (possibly due to an GitHub iOS App bugs...). Sorry for the confusion!

source code contributions will slowly turn to [LGPL v2.1+][LGPL v2.1].

### 3. A LGPL v3+ package, and an LGPL v2.1+ source code (Optional)

Should we obtain approval to relicense from our developers before we remove our [Apache 2.0][]
dependencies, we will then start releasing the MDAnalysis package under a [LGPL v3+][LGPL v3] license.

### 4. An LGPL v2.1+ package and source code

This is the state which we aim the library to eventually be licensed under. Once we both obtain
the agreement of our developers to relicense and remove [Apache 2.0][] dependencies, we will be
able to release the MDAnalysis package under [LGPL v2.1+][].

## How does this affect me?

> 

Reviewer: How?

Author: Not sure I understand what you mean here.

Reviewer: This was meant for the paragraph below, namely "you will be asked to relicense them".

Author: Adding link to #contacting-contributors

### As a contributor

If you have any historical contributions you will be asked to relicense them to [LGPL v2.1+][LGPL v2.1]
instead of the initially proposed [LGPL v3+][LGPL v3].

### As a developer

Ultimately we believe that the proposed change to [LGPL v2.1+][LGPL v2.1] should offer a better compatibility
(Table 1) with whatever license you may choose to use for your own code. The interim package license changes
may however impact you, if you believe this to be the case [please do get in touch with us](#getting-in-touch).

### As a user

> We do not anticipate this to have a significant impact on users. There may be some slight changes to how you

Reviewer: How? I think I mentioned this before, but we might take inspiration from https://github.com/spack/spack/issues/9144 (if we haven't already).

Reviewer: More specifically: https://github.com/spack/spack/issues/9137

Author: I think this comment is linking to the wrong place?

Reviewer: Oh, shit. Looks like all of them are off...

Bug in the GitHub iOS app, most likely...

Author: I got the idea of most of them :)

Reviewer: This was meant for the section below:

> We will shortly be contacting every historical contributor to MDAnalysis and ask them to agree to changing
their contribution’s license from the current “GPL v2 or any later version” to “LGPL v2.1 or any later version”.

Really sorry for the confusion.

[might access MMTF files which are discussed below](#removing-apache-20-dependencies).

## Next steps

> ### Contacting contributors

Reviewer: What if the contributor is not the copyright owner? Should we provide a more general statement that the contributor confirms that the copyright owner agrees to the change? (I'm thinking of cases where someone contributes but the copyright is technically with the employer).

Reviewer: For examples, this is what was written in Spack:

> By granting this permission, you are confirming that you have the right to make your contribution(s) to Spack and to grant this permission, and that if you made your contribution(s) in the course of your employment, that such contributions and consent have been authorized by your employer.

Reviewer: See https://github.com/spack/spack/issues/9137

Author: The wording we have is lawyer approved already - we're not changing from it.

The question on being the copyright holder is right to ask but also very complicated legally and not something I would address here - I'll add a comment about asking us if they have any concerns about it.

At the end of the day this is lawyer realm. and we might have to go back to NF with questions.

Reviewer: Well, it think this might be a problem for people that contributed while working for an employer that holds the copyright. The phrasing we have says specifically "I am the copyright owner", and does not account for cases where the contributor isn't (even if he has the consent of their employer for the relicensing). (Obligatory disclaimer: I'm not a lawyer.)

Author: All I can say here is "we know" - but this is lawyers orders.

Author: See what I sent you on discord :)

Author: I've added a note about being the copyright onwer - as per the intended text of the full email, we'll just ask folks to let us know if they don't know and we will just have to workshop it with them. My hope is that for our major contributors we can get blanket exceptions from say Manchester, Oxford, ASU, etc..

Reviewer: Blanket exceptions would be great. I think the note looks good.

We will shortly be contacting every historical contributor to MDAnalysis and ask them to agree to changing
their contribution’s license from the current “GPL v2 or any later version” to “LGPL v2.1 or any later version”.

Specifically we will be asking all contributors to respond with the following statement:

```
I, <your name>, GitHub handle <your GitHub handle>, am the copyright owner of all the code I submitted
to the MDAnalysis library using my GitHub handle. I agree to relicense all my code contributions to the
MDAnalysis library under the LGPLv2.1+ license from this day going forward.

Typed Signature
```

> It is important that we hear back from as many contributors as possible. If you have contributed to

Reviewer: Should these be moved near the top? I think it could make for a much easier reading of the blog post to clarify these points at the beginning.

Author: I don't think we can bypass the "why" before we get to the "how" - could add a TOC though?

MDAnalysis in the past but have since changed your git-linked contact details, we would kindly ask if
you could email *licensing@mdanalysis.org* to let us know how best to contact you.

### Removing Apache 2.0 dependencies

As we move towards a v3.0 release of MDAnalysis, we will be looking to replace existing [Apache 2.0][]
core dependencies. Of particular impact, this may mean that access to reading [MMTF][] files may no longer
be available by default through the MDAnalysis library, but will instead be offered through an optionally
installable [MDAKit](https://mdakits.mdanalysis.org/about.html).

## Getting in touch

We understand that license changes are a complex subject that can have a large impact on how users and
developers may interact with the MDAnalysis library. Should you have any concerns, questions, or comments,
we urge you to get in touch with the core developer team. You can do so over our
[public forums[({% link index.md %}#participating) or privately by emailing *licensing@mdanalysis.org*.

> ## Definitions

Reviewer: Should these be moved near the top? I think it could make for a much easier reading of the blog post to clarify these points at the beginning.

Author: I had a comment about this elsewhere, about how we shouldn't do the "how" before we do the "why" but that a TOC might fix this. 

This is all to say that I've added a TOC.

> Title: An update on relicensing the MDAnalysis library (towards LGPLv2.1+)

Reviewer 2: ```suggestion
---
layout: post
title: An update on relicensing the MDAnalysis library (towards LGPLv2.1+)
---
```

> [!WARNING]
> **Disclaimer**
> The MDAnalysis core team members are not
> lawyers. As such the information provided here __does not, and is not
> intended to constitute legal advice__. This blog post also does not
> represent MDAnalysis' full legal position on software licensing; it
> simply aims to inform MDAnalysis developers and users on certain
> aspects of the MDAnalysis relicensing efforts and how we think it
> may impact them going forward.
>
> Further information on open-source software licensing can be found
> from sources such as the [Open Source Initiative][OSI],
> [tl;dr Legal][tldr legal] and the [Software Sustainability Institute][SSI licensing].
>
> Should you have any concerns about licensing, we always strongly
> recommend getting legal advice before making any decisions on how
> licensing changes may affect you.

## Short description

This blog post provides an update on MDAnalysis’ proposed license change. Specifically, we outline our
decision to opt for a [GNU Lesser General Public License v2.1+ (LGPL v2.1+)][LGPL v2.1] instead of

> the [previously communicated]() [GNU Lesser General Public License v3+ (LGPL v3+)][LGPL v3].

Reviewer 3: missing link?

[upcoming actions for library contributors](#package-license).

## Overview

In November of 2022 the MDAnalysis team [announced a proposal to change the license of its core library](https://www.mdanalysis.org/2022/11/07/relicensing/#relicensing-mdanalysis)
from the [GNU General Public License v2+][GPL v2] to the [GNU Lesser General Public License v3+][LGPL v3].
The full rationale for this change can be found [here](https://www.mdanalysis.org/2022/11/07/relicensing/#rationale-for-license-change)
and mainly centered around three main arguments:

1. Allowing downstream codes to freely import or link to the MDAnalysis library with minimal worry about the license they choose to use.
2. Allowing downstream codes to use and subclass any MDAnalysis components under its application programming interface
   (classes, methods, and data objects), without incurring copyleft and incurring the license choice of the downstream code.
3. Ensuring that codes that either directly copy or extend the MDAnalysis library’s source code would fall under the copyleft
   license requirements of the MDAnalysis library license. Doing so would ensure that eventual license would more closely match the

>    original intent of the historical [GPL v2+](GPL v2) contributions which were kindly donated to MDAnalysis over the years.

Reviewer 3: ```suggestion
   original intent of the historical [GPL v2+][GPL v2] contributions which were kindly donated to MDAnalysis over the years.
```

| [MIT][]         | Yes ([copyleft]())  | Yes ([copyleft]())  | Yes                 | Yes                 | Yes                | Yes                | Yes                |
| [BSD 3.0][]     | Yes ([copyleft]())  | Yes ([copyleft]())  | Yes                 | Yes                 | Yes                | Yes                | Yes                |

*\[1\] Compatible if the [GPL v2][] license is provided under the “or any later version” terms.
\[2\] Compatible if the [LGPL v2.1][] license is provided under the “or any later version” terms.*

### Why LGPL v2.1+ instead of LGPLv3+?

During the community consultation, it was brought to our attention that by choosing an [LGPL v3+][LGPL v3]
license, would no longer be compatible with older versions of the GNU General Public License, namely
[GPL v2][] and [LGPL v2.1][] (see Table 1). The reasons for this are complex and we will not attempt to
explain them here, we will instead refer the reader to the [FSF GPL license compatibility FAQ entry](https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility)
for further details.

Overall the impact of this incompatibility is likely very low. Few codes are solely licensed under

> [GPLv2][] or [LGPLv2.1][], with most under “or any later version” terms, allowing the codes to relicense

Reviewer 3: ```suggestion
[GPL v2][] or [LGPL v2.1][], with most under “or any later version” terms, allowing the codes to relicense
```

[MMTF][] files in favor [newer PDB file formats](https://mmcif.wwpdb.org/). As this will likely take some time,
we will temporarily change our [package license](#package-license) to [GPL v3+][GPL v3], which we have already done
as part of the [2.6.0 release of MDAnalysis](https://www.mdanalysis.org/blog/#release-2-6-0-and-2-6-1-of-mdanalysis).
We may also possibly interim [LGPL v3+][LGPL v3] license, before reaching our final [LGPL v2.1+][LGPL v2.1]
[package license](#package-license). Further details how we see the license process to occur can be found
[later in this blog](#detailing-our-proposed-relicensing-process).

We note that we can make these changes to the MDAnalysis [package license](#package-license) without requiring
sign offs from all developers as the library’s [source code license](#source-code-license) will still remain
under [GPL v2+][GPL v2] (and eventually [LGPL v2.1+][LGPL v2.1]). This allows us to release the package under
any appropriate license version of the GNU GPL. It does however mean that we will be temporarily at risk of
downstream packages encountering the [previously mentioned incompatibilities](#why-lgpl-v21-instead-of-lgplv3)

> between [GPL v2][] and [GPLv3][]. **Should this impact you, please [get in touch with the core developer team](#getting-in-touch)

Reviewer 3: ```suggestion
between [GPL v2][] and [GPL v3][]. **Should this impact you, please [get in touch with the core developer team](#getting-in-touch)
```

As of commit [44733fc][], all new contributions to MDAnalysis have been made under the terms of the
[LGPL v2.1+][LGPL v2.1] and the package is released under [GPL v3+][GPL v3] to better reflect our
understanding of the [Apache 2.0 incompatibility issues](#why-the-interim-gpl-v3-and-lgpl-v3-package-licenses-the-issue-with-apache-20).
Historical source code contributions remain under the terms of [GPL v2+][GPL v2].

This will remain the state of MDAnalysis until we obtain approval from our developers to relicense
their code to [LGPL v2.1+][LGPL v2.1]. As we obtain contributor agreements, these historical
source code contributions will slowly turn to [LGPL v2.1+][LGPL v2.1].

### 3. A LGPL v3+ package, and an LGPL v2.1+ source code (Optional)

Should we obtain approval to relicense from our developers before we remove our [Apache 2.0][]
dependencies, we will then start releasing the MDAnalysis package under a [LGPL v3+][LGPL v3] license.

### 4. An LGPL v2.1+ package and source code

This is the state which we aim the library to eventually be licensed under. Once we both obtain
the agreement of our developers to relicense and remove [Apache 2.0][] dependencies, we will be

> able to release the MDAnalysis package under [LGPL v2.1+][].

Reviewer 3: ```suggestion
able to release the MDAnalysis package under [LGPL v2.1+][LGPL v2.1].
```

MDAnalysis library under the LGPLv2.1+ license from this day going forward.

Typed Signature
```

It is important that we hear back from as many contributors as possible. If you have contributed to
MDAnalysis in the past but have since changed your git-linked contact details, we would kindly ask if
you could email *licensing@mdanalysis.org* to let us know how best to contact you.

### Removing Apache 2.0 dependencies

As we move towards a v3.0 release of MDAnalysis, we will be looking to replace existing [Apache 2.0][]
core dependencies. Of particular impact, this may mean that access to reading [MMTF][] files may no longer
be available by default through the MDAnalysis library, but will instead be offered through an optionally
installable [MDAKit](https://mdakits.mdanalysis.org/about.html).

## Getting in touch

We understand that license changes are a complex subject that can have a large impact on how users and
developers may interact with the MDAnalysis library. Should you have any concerns, questions, or comments,
we urge you to get in touch with the core developer team. You can do so over our

> [public forums[({% link index.md %}#participating) or privately by emailing *licensing@mdanalysis.org*.

Reviewer 3: ```suggestion
[public forums]({% link index.md %}#participating) or privately by emailing *licensing@mdanalysis.org*.
```

## Acknowledgements

We are very grateful for the administrative and legal support from our fiscal sponsor, [NumFOCUS][].

This relicensing effort has been made possible in part by the CZI grant DAF2021-237663,
grant DOI https://doi.org/10.37921/426590wiobus from the Chan Zuckerberg Initiative DAF,
an advised fund of Silicon Valley Community Foundation (funder DOI 10.13039/100014989).

– The MDAnalysis [Core Developers][]

[OSI]: https://opensource.org/osd
[tldr legal]: https://tldrlegal.com/
[SSI licensing]: https://www.software.ac.uk/resources/guides/choosing-open-source-licence
[OSD]: https://opensource.org/osd
[GPL v2]: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
[GPL v3]: https://www.gnu.org/licenses/gpl-3.0.en.html
[LGPL v2.1]: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
[LGPL v3]: https://www.gnu.org/licenses/lgpl-3.0.en.html
[MIT]: https://opensource.org/license/mit/
[BSD 3.0]: https://opensource.org/license/bsd-3-clause/
[Apache 2.0]: https://www.apache.org/licenses/LICENSE-2.0
[contributors]: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/AUTHORS
[NumFOCUS]: https://www.numfocus.org

> [expert opinions]: https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=heather+meeker+open+source+for+business&btnG=

Reviewer 3: would it maybe be better to cite direct rather than use a google scholar link? (especially since it seems google scholar doesn't actually link you to anywhere to read this?)

Author: Couldn't really work this one out, so I just referenced the book in the text directly.

Author: Couldn't really work this one out, so I just referenced the book in the text directly.

---
layout: post
title: An update on relicensing the MDAnalysis library (towards LGPLv2.1+)
---

> ⚠️ **Disclaimer**
> The MDAnalysis core team members are not
> lawyers. As such the information provided here __does not, and is not
> intended to constitute legal advice__. This blog post also does not
> represent MDAnalysis' full legal position on software licensing; it
> simply aims to inform MDAnalysis developers and users on certain
> aspects of the MDAnalysis relicensing efforts and how we think it
> may impact them going forward.
>
> Further information on open-source software licensing can be found
> from sources such as the [Open Source Initiative][OSI],
> [tl;dr Legal][tldr legal] and the [Software Sustainability Institute][SSI licensing].
>
> Should you have any concerns about licensing, we always strongly
> recommend getting legal advice before making any decisions on how
> licensing changes may affect you.

> ## Table of contents

Reviewer: Very helpful, thanks @IAlibay

- [Short description](#short-description)
- [Overview](#overview)
- [Rationale for our change in target license](#rationale-for-our-change-in-target-license)

>   * [Why LGPL v2.1+ instead of LGPLv3+?](#why-lgpl-v21--instead-of-lgplv3--)

Reviewer 3: ```suggestion
  * [Why LGPL v2.1+ instead of LGPLv3+?](#why-lgpl-v21-instead-of-lgplv3)
```

>   * [Why the interim GPL v3+ and LGPL v3+ package licenses? (The issue with Apache 2.0).](#why-the-interim-gpl-v3--and-lgpl-v3--package-licenses---the-issue-with-apache-20-)

Reviewer 3: ```suggestion
  * [Why the interim GPL v3+ and LGPL v3+ package licenses? (The issue with Apache 2.0).](#why-the-interim-gpl-v3-and-lgpl-v3-package-licenses-the-issue-with-apache-20)
```

>   * [Why not opt for a more permissive (and likely less complicated) license such as MIT?](#why-not-opt-for-a-more-permissive--and-likely-less-complicated--license-such-as-mit-)

Reviewer 3: ```suggestion
  * [Why not opt for a more permissive (and likely less complicated) license such as MIT?](#why-not-opt-for-a-more-permissive-and-likely-less-complicated-license-such-as-mit)
```

- [Detailing our proposed relicensing process](#detailing-our-proposed-relicensing-process)

>   * [1. A GPL v2+ package and source code (MDAnalysis v2.5 and lower)](#1-a-gpl-v2--package-and-source-code--mdanalysis-v25-and-lower-)

Reviewer 3: ```suggestion
  * [1. A GPL v2+ package and source code (MDAnalysis v2.5 and lower)](#1-a-gpl-v2-package-and-source-code-mdanalysis-v25-and-lower)
```

>   * [2. A GPL v3+ package, and a mix of historical GPL v2+ and new LGPL v2.1+ source code (MDAnalysis v2.6 onwards)](#2-a-gpl-v3--package--and-a-mix-of-historical-gpl-v2--and-new-lgpl-v21--source-code--mdanalysis-v26-onwards-)

Reviewer 3: ```suggestion
  * [2. A GPL v3+ package, and a mix of historical GPL v2+ and new LGPL v2.1+ source code (MDAnalysis v2.6 onwards)](#2-a-gpl-v3-package-and-a-mix-of-historical-gpl-v2-and-new-lgpl-v21-source-code-mdanalysis-v26-onwards)
```

>   * [3. An LGPL v3+ package, and an LGPL v2.1+ source code (Optional)](#3-an-lgpl-v3--package--and-an-lgpl-v21--source-code--optional-)

Reviewer 3: ```suggestion
  * [3. An LGPL v3+ package, and an LGPL v2.1+ source code (Optional)](#3-an-lgpl-v3-package-and-an-lgpl-v21-source-code-optional)
```

>   * [4. An LGPL v2.1+ package and source code](#4-an-lgpl-v21--package-and-source-code)

Reviewer 3: ```suggestion
  * [4. An LGPL v2.1+ package and source code](#4-an-lgpl-v21-package-and-source-code)
```

> - [How does this affect me?](#how-does-this-affect-me-)

Reviewer 3: ```suggestion
- [How does this affect me?](#how-does-this-affect-me)
```

After careful consideration we have decided to aim for a license change to the [LGPL v2.1+][LGPL v2.1] via interim
[package license](#package-license) changes to [GPL v3+][GPL v3] and [LGPL v3+][LGPL v3].

In this post we will provide our rationale for this change, alongside key information for how we will be
enacting these changes.

**License changes will affect how users and developers will interact with the MDAnalysis code base going
forward. We urge all members of the community to read this post and [get in touch](#getting-in-touch) with
the MDAnalysis core developer team should you have any questions or concerns**.

## Rationale for our change in target license

*Table 1: MDAnalysis’ view of the dynamic linking license compatibility matrix. License pairs
detailed with a compatibility of “No” have claimed incompatibilities between their license terms.
License pairs detailed as “Yes ([copyleft](#copyleft))” are compatible but may incur copyleft through the
interacting GNU GPL license. License pairs detailed with the compatibility “Yes” are, to our knowledge,
fully compatible without incurring any [copyleft](#copyleft).*

> | License         | [GPL v2][]                   | [GPL v3][]                   | [LGPL v2.1][]                | [LGPL v3][]                  | [Apache 2.0]()              | [MIT][]                     | [BSD 3.0][]                  |

Reviewer 3: ```suggestion
| License         | [GPL v2][]                   | [GPL v3][]                   | [LGPL v2.1][]                | [LGPL v3][]                  | [Apache 2.0][]              | [MIT][]                     | [BSD 3.0][]                  |
```

## Definitions

### package license

The license under which the MDAnalysis library is packaged and released. This may be different from
but still compatible with the [source code license](#source-code-license).

### source code license

The license under which the MDAnalysis source code is released. This only accounts for the code itself
and not any impact in licensing incurred from linking (i.e. `python importing`) to other libraries.

### license compatibility

Whether or not the terms of two licenses allows them to be compatible with each other.

### copyleft

Whether the terms of a given license enforces that other codes (even under other licenses) must be bound
by the same conditions. See the [FSF definition](https://www.gnu.org/licenses/copyleft.en.html) for more information.

## Acknowledgements

We are very grateful for the administrative and legal support from our fiscal sponsor, [NumFOCUS][].

This relicensing effort has been made possible in part by the CZI grant DAF2021-237663,
grant DOI https://doi.org/10.37921/426590wiobus from the Chan Zuckerberg Initiative DAF,
an advised fund of Silicon Valley Community Foundation (funder DOI 10.13039/100014989).

> – Irfan Alibay on behalf of the MDAnalysis [Core Developers][]

Author: I'm maybe being a bit much here, so I'm willing to hear what folks think / not do this. At the end of the day this has been much of a solo project that has cost me far far too much time (just this blog alone was order 15h combined with checking up legal stuff). Unlike a repo I can't really put my name on anything else but this.

Reviewer 4: I am cool with you signing it on the behalf of the core devs.

Reviewer 3: I'd been thinking to suggest something you do this, so I'm definitely in support of having your name there!

[GPL v2]: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
[GPL v3]: https://www.gnu.org/licenses/gpl-3.0.en.html
[LGPL v2.1]: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
[LGPL v3]: https://www.gnu.org/licenses/lgpl-3.0.en.html
[MIT]: https://opensource.org/license/mit/
[BSD 3.0]: https://opensource.org/license/bsd-3-clause/
[Apache 2.0]: https://www.apache.org/licenses/LICENSE-2.0
[contributors]: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/AUTHORS
[NumFOCUS]: https://www.numfocus.org
[fasteners]: https://fasteners.readthedocs.io/en/latest/
[mmtf-python]: https://github.com/rcsb/mmtf-python
[py-filelock]: https://github.com/tox-dev/filelock
[MMTF]: https://mmtf.rcsb.org/
[44733fc]: https://github.com/MDAnalysis/mdanalysis/commit/44733fc214dcfdcc2b7cb3e3705258781bb491bd
[MDAnalysis core library]: https://github.com/MDAnalysis/mdanalysis
[previous relicensing post]: {{ site.baseurl }}{% post_url 2022-11-07-relicensing %}
[rationale]: {{ site.baseurl }}{% post_url 2022-11-07-relicensing %}#rationale-for-license-change
[consultation]:  {{ site.baseurl }}{% post_url 2022-11-07-relicensing %}#consultation-period-7th-november-until-5th-december-2022

> [release 2.6]: {{ site.baseurl }}{% post_url 2023-09-02-release-2.6.1.md %}

Reviewer 2: ```suggestion
[release 2.6]: {{ site.baseurl }}{% post_url 2023-09-02-release-2.6.1 %}
```

Fixes render:
![Uploading Screenshot 2023-09-25 at 12.45.06 pm.png…]()

Author: Yeah sorry that's my bad - silly copy paste typo. I still can't get jekyll to behave on my laptop so local testing isn't happening.
